workers/main/src/common/Redmine.ts

```diff
@@ -0,0 +1,75 @@
+import * as mysql from 'mysql2/promise';
+import { Pool, PoolOptions, RowDataPacket } from 'mysql2/promise';
+
+import { ProjectUnit } from './types';
+
+export class Redmine {
+  private pool: Pool;
+  private credentials: PoolOptions;
+  private poolEnded = false;
+
+  constructor(credentials: PoolOptions) {
+    this.credentials = credentials;
+    this.pool = mysql.createPool(this.credentials);
+  }
+
+  private ensureConnection() {
+    if (!this.pool || this.poolEnded) {
+      this.pool = mysql.createPool(this.credentials);
+      this.poolEnded = false;
+    }
+  }
+
+  /** Call this when you want to end the pool */
+  async endPool() {
+    if (this.pool && !this.poolEnded) {
+      await this.pool.end();
+      this.poolEnded = true;
+    }
+  }
+
+  private getProjectUnitsQuery(options?: {
```

Comment:
Why don you need optional? I prefer not having optional types and make code as explicit as possible